bridge/svix-bridge-plugin-queue/src/config.rs

```diff
@@ -0,0 +1,88 @@
+pub use crate::gcp_pubsub::{GCPPubSubInputOpts, GCPPubSubOutputOpts};
+pub use crate::rabbitmq::{RabbitMqInputOpts, RabbitMqOutputOpts};
+pub use crate::receiver_output::QueueForwarder;
+pub use crate::redis::{RedisInputOpts, RedisOutputOpts};
+pub use crate::sqs::{SqsInputOpts, SqsOutputOpts};
+use crate::{
+    GCPPubSubConsumerPlugin, RabbitMqConsumerPlugin, RedisConsumerPlugin, SqsConsumerPlugin,
+};
+use serde::Deserialize;
+use svix_bridge_types::{ReceiverOutput, SenderInput, SenderOutputOpts};
+
+#[derive(Deserialize)]
+pub struct QueueConsumerConfig {
+    pub name: String,
+    pub input: SenderInputOpts,
+    #[serde(default)]
+    pub transformation: Option<String>,
+    pub output: SenderOutputOpts,
+}
+
+impl TryInto<Box<dyn SenderInput>> for QueueConsumerConfig {
```

Comment:
Does this even need to be `TryInto`? It looks like it's infallible, no?

Reply:
Artifact of churn. Originally my thinking was we'd use `try_into()` as the de facto form of constructing a "plugin" from the corresponding config in a general way, and allow for plugins to do last-ditch validation of the config (leaning towards the fallible).

This fell apart pretty quickly when the producer-side needed to be constructed async (and `TryInto` isn't `async`).
The second way it fell apart was I think we are largely needing to branch by concrete type anyway so having `TryInto` as a general interface is not needed.

Anyway, I'll look to prune this out in favor of a dedicated constructor/helper. :+1:

Reply:
Updated :vulcan_salute: